core/src/main/2.0/scala/com/holdenkarau/spark/testing/DataFrameSuiteBase.scala

```diff
@@ -254,10 +255,16 @@ trait DataFrameSuiteBaseLike extends SparkContextProvider
     * Compares if two [[DataFrame]]s are equal, checks that the schemas are the same.
     * When comparing inexact fields uses tol.
     *
-    * @param tol max acceptable tolerance, should be less than 1.
+    * @param tol          max acceptable decimal tolerance, should be less than 1.
+    * @param tolTimestamp max acceptable timestamp tolerance.
```

Comment:
This seems like a good improvement, in the release notes we should also make note of this since before `tol` was used for timestamp tolerance as well.
We could do the hacky thing where we copy tol over to tolTimestamp for default but I think this is fine provided we make a note of it.

Reply:
I didn't change this methods because they use default values, so (I believe) they shouldn't break anything.

If no break change applies we could use a minor bump? I used 1.5.0 version for release notes

check it out and let me know if this solves the issue

Reply:
So while it's not a breaking API change in that the code will still compile, it is sort of a breaking semantic change (e.g. if you been using `tol` for timestamp comparison your tests will now fail). That being said it's probably ok for a minor version change called out in the release notes.

For the description of the `tol` parameter we should say numeric rather than decimal, we use the same tolerance value for float & double as well.

Reply:
what about using the same approach as the other methods and deprecating the "old" method?:

```
  /**
    * Compares if two [[DataFrame]]s are equal, checks that the schemas are the same.
    * When comparing inexact fields uses tol.
    *
    * @param tol          max acceptable tolerance for numeric (between(0, 1)) &
    *                     timestamp (millis).
    * @param customShow   unit function to customize the '''show''' method
    *                     when dataframes are not equal. IE: '''df.show(false)''' or
    *                     '''df.toJSON.show(false)'''.
    */
  @deprecated(
    "Use `assertDataFrameApproximateEquals` with timestamp tolerance",
    since = "1.5.0"
  )
  def assertDataFrameApproximateEquals(
    expected: DataFrame, result: DataFrame,
    tol: Double, customShow: DataFrame => Unit = _.show()): Unit =
    assertDataFrameApproximateEquals(expected, result, tol,
      Duration.ofNanos((tol * 1000).toLong), customShow)

  /**
    * Compares if two [[DataFrame]]s are equal, checks that the schemas are the same.
    * When comparing inexact fields uses tol & tolTimestamp.
    *
    * @param tol          max acceptable numeric tolerance, should be less than 1.
    * @param tolTimestamp max acceptable timestamp tolerance.
    * @param customShow   unit function to customize the '''show''' method
    *                     when dataframes are not equal. IE: '''df.show(false)''' or
    *                     '''df.toJSON.show(false)'''.
    */
  def assertDataFrameApproximateEquals(
    expected: DataFrame, result: DataFrame,
    tol: Double, tolTimestamp: Duration,
    customShow: DataFrame => Unit = _.show()): Unit = {.................}
```

Reply:
I like it :)